e2e/test/iothub/DeviceTokenRefreshE2ETests.cs

```diff
@@ -109,6 +109,63 @@ public async Task DeviceClient_TokenConnectionDoubleRelease_Ok()
             } // Second release
         }
 
+        // The easiest way to test that sas tokens expire with custom expiration time via the CreateFromConnectionString flow is 
+        // by initializing a DeviceClient instance over Mqtt (since sas token expiration over Mqtt is accompanied by a disconnect).
```

Comment:
This does have the possibility of giving us false positives, as in, a network issue could also give us a disconnect. Since the token refresh logic is encapsulated entirely within the sdk, we cannot monitor it from outside.
The translation of client options to the token refresher also happens over multiple layers, so this wasn't a good candidate for unit tests either.

Let me know if you have any suggestions for testing this in a different way.

Reply:
Is this something we can handle and track in long haul?

Reply:
We could set a custom token expiration time and monitor if it disconnects as per the schedule over an extended period of time. The core token refresh logic is tested separately within our test suite, what we need to test here is to ensure that the client options are being set into the token refresher class.
This does fit into a unit test better than an E2E test, it is just the way these options are translated in our client that doesn't make it straightforward to test. Nevertheless, that's where I am working on adding a test now.